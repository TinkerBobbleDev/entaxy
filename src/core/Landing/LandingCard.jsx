import React from 'react'
import { useSelector } from 'react-redux'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Paper from '@material-ui/core/Paper'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import Avatar from '@material-ui/core/Avatar'
import grey from '@material-ui/core/colors/grey'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import CardActions from '@material-ui/core/CardActions'
import PowerSettingsNewIcon from '@material-ui/icons/PowerSettingsNew'
import { loginAs, userLogout } from '../../store'
import blockstackLogo from './blockstack-bug-rounded.svg'
import LinkTo from '../../common/LinkTo'

const useStyles = makeStyles((theme) => ({
  loggedOutContainer: {
    padding: theme.spacing(1),
    background: grey[200]
  },
  signinButton: {
    width: '100%',
    marginBottom: theme.spacing(3)
  },
  blockstackTitle: {
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'left',
    backgroundSize: '15px',
    backgroundImage: `url(${blockstackLogo})`,
    paddingLeft: '20px',
    '& a': {
      color: theme.palette.action.active
    }
  },
  browserIcon: {
    fontSize: 19,
    marginBottom: -4,
    marginRight: 10
  },
  logoutIcon: {
    fontSize: 16,
    marginRight: theme.spacing(1)
  },
  logoutButton: {
    marginLeft: 'auto',
    whiteSpace: 'nowrap'
  }
}))

export const LandingCardComponent = ({ history }) => {
  const classes = useStyles()
  const user = useSelector((state) => state.user)

  const login = (loginType) => {
    loginAs(loginType)
    history.push('/dashboard')
  }

  if (user.isAuthenticatedWith) {
    return (
      <Grid container justify="center">
        <Grid item xs={12} sm={8} md={6} lg={4}>
          <Card>
            <CardHeader
              avatar={<Avatar src={user.pictureUrl} alt={user.name} />}
              title={user.name}
            />
            <CardActions disableSpacing>
              <Button
                size="small"
                color="secondary"
                component={LinkTo('/dashboard')}
              >
                Continue to your Dashboard
              </Button>
              <Button
                size="small"
                color="secondary"
                onClick={userLogout}
                className={classes.logoutButton}
              >
                <PowerSettingsNewIcon className={classes.logoutIcon} />
                Sign out
              </Button>
            </CardActions>
          </Card>
        </Grid>
      </Grid>
    )
  }

  return (
    <Grid container spacing={2} justify="center">
      <Grid item xs={12} md={4}>
        <Paper elevation={1} className={classes.loggedOutContainer}>
          <Button
            variant="contained"
            color="secondary"
            className={classes.signinButton}
            onClick={() => login('blockstack')}
            data-testid="signinWithBlockstackButton"
          >
            Sign in with Blockstack
          </Button>
          <Typography variant="caption" paragraph align="center">
            Keep your data encrypted and decentralized.
            <br />Learn more about &nbsp;
            <span className={classes.blockstackTitle}>
              <a href="https://blockstack.org/what-is-blockstack" target="_blank" rel="noopener noreferrer">
                Blockstack
              </a>
            </span>
          </Typography>
        </Paper>
      </Grid>
      <Grid item xs={12} md={4}>
        <Paper elevation={1} className={classes.loggedOutContainer}>
          <Button
            variant="contained"
            color="secondary"
            className={classes.signinButton}
            onClick={() => login('guest')}
            data-testid="signinAsGuestButton"
          >
            Sign in as guest user
          </Button>
          <Typography variant="caption" paragraph align="center">
            Your data will be stored on this browser only.
            <br />Great for exploring the app.
          </Typography>
        </Paper>
      </Grid>
    </Grid>
  )
}

LandingCardComponent.propTypes = {
  history: PropTypes.object.isRequired
}

export default LandingCardComponent
