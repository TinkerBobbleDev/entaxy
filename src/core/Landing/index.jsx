import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import grey from '@material-ui/core/colors/grey'
import LandingCard from './LandingCard'
import landingImage from './landing.png'

const styles = () => ({
  root: {
    display: 'flex',
    height: '100vh',
    padding: '20px'
  },
  left: {
    'align-items': 'left',
    'justify-content': 'space-evenly',
    background: grey[100],
    padding: '2% 5% 2% 10%',
    display: 'flex',
    'flex-flow': 'column nowrap'
  },
  logo: {
    font: 'bold 24px var(--font-garden-grove)',
    color: 'black'
  },
  tagline: {
    font: 'italic 11px var(--font-garden-grove)',
    color: grey[500],
    'padding-top': '3px'
  },
  title: {
    font: 'bold 48px var(--font-garden-grove)',
    'line-height': '1.2'

  },
  description: {
    font: '18px var(--font-garden-grove)',
    'padding-top': '15px',
    'line-height': '1.6'

  },
  cards: {
    display: 'flex',
    'flex-direction': 'column'
  },
  right: {
    'background-repeat': 'no-repeat',
    'background-position': 'center',
    'background-size': '100%',
    'background-image': `url(${landingImage}), linear-gradient(to bottom, var(--color-blue), var(--color-cyan))`
  }
})

const Landing = ({ classes }) => (
  <Grid container spacing={0} className={classes.root}>
    <Grid item xs={6} className={classes.left}>
      <div className={classes.logo}>
        Entaxy
        <div className={classes.tagline}>Order from chaos</div>
      </div>
      <div className={classes.title}>
        Your Personal Finance Simple & Private
        <div className={classes.description}>
          Insights into your finances,
          without sacrificing your data
        </div>
      </div>
      <Grid container className={classes.cards}>
        <LandingCard
          title="Income Taxes"
          description="At some point you will wonder how they work"
          path="/taxes"
        />
        <LandingCard
          title="Portfolio"
          description="ETFs, Crypto, etc"
          path="/portfolio"
        />
      </Grid>
    </Grid>
    <Grid item xs={6} className={classes.right} />
  </Grid>
)

Landing.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(Landing)
