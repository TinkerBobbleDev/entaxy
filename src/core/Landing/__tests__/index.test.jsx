import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Landing from '../'

describe('LandingCard', () => {
  it('matches snapshot', () => {
    const props = {
      classes: {}
    }
    const component = renderer.create((
      <BrowserRouter>
        <Landing props={{ ...props }} />
      </BrowserRouter>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
