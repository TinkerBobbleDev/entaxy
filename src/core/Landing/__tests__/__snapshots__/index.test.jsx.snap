// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCard matches snapshot 1`] = `
<div
  className="MuiGrid-container-9 Landing-root-1"
>
  <div
    className="MuiGrid-item-10 MuiGrid-grid-xs-6-41 Landing-left-2"
  >
    <div
      className="Landing-logo-3"
    >
      Entaxy
      <div
        className="Landing-tagline-4"
      >
        Order from chaos
      </div>
    </div>
    <div
      className="Landing-title-5"
    >
      Your Personal Finance Simple & Private
      <div
        className="Landing-description-6"
      >
        Insights into your finances, without sacrificing your data
      </div>
    </div>
    <div
      className="MuiGrid-container-9 Landing-cards-7"
    >
      <div
        className="MuiPaper-root-103 MuiPaper-elevation2-107 MuiPaper-rounded-104 MuiCard-root-102 LandingCard-card-100"
      >
        <div
          className="MuiCardHeader-root-130"
        >
          <div
            className="MuiCardHeader-content-133"
          >
            <span
              className="MuiTypography-root-136 MuiTypography-headline-141 MuiCardHeader-title-134"
            >
              Income Taxes
            </span>
            <span
              className="MuiTypography-root-136 MuiTypography-body1-145 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-135"
            >
              At some point you will wonder how they work
            </span>
          </div>
          <div
            className="MuiCardHeader-action-132"
          >
            <a
              className="MuiButtonBase-root-166 MuiIconButton-root-160 MuiIconButton-colorSecondary-163 LandingCard-cardButton-101"
              href="/taxes"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
              variant="fab"
            >
              <span
                className="MuiIconButton-label-165"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-169"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-175"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root-103 MuiPaper-elevation2-107 MuiPaper-rounded-104 MuiCard-root-102 LandingCard-card-100"
      >
        <div
          className="MuiCardHeader-root-130"
        >
          <div
            className="MuiCardHeader-content-133"
          >
            <span
              className="MuiTypography-root-136 MuiTypography-headline-141 MuiCardHeader-title-134"
            >
              Portfolio
            </span>
            <span
              className="MuiTypography-root-136 MuiTypography-body1-145 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-135"
            >
              ETFs, Crypto, etc
            </span>
          </div>
          <div
            className="MuiCardHeader-action-132"
          >
            <a
              className="MuiButtonBase-root-166 MuiIconButton-root-160 MuiIconButton-colorSecondary-163 LandingCard-cardButton-101"
              href="/portfolio"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
              variant="fab"
            >
              <span
                className="MuiIconButton-label-165"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-169"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-175"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-10 MuiGrid-grid-xs-6-41 Landing-right-8"
  />
</div>
`;
